### ISSUE TITLE
Allow pinning of buffers

### ISSUE THREAD

**Issue by leath-dub** [leath-dub/snipe.nvim#44]
I am thinking about adding a kind of harpoon like feature where you can pin buffers/files and save them after close.

Maybe this list is in a separate menu but you can also set an option to auto open those pinned files so they are accessible in the main menu.

Feel free to add musings/thoughts below.

**Comment by omerdn1** [comment]
Love this! I'd like to see an implementation that utilizes the native session API instead of saving the routes to file, which is how harpoon has been doing things, and it proved to be slow at scale.

**Comment by leath-dub** [comment]
> Love this! I'd like to see an implementation that utilizes the native session API instead of saving the routes to file, which is how harpoon has been doing things, and it proved to be slow at scale.

I have not heard of this ! gonna look into this.

**Comment by leath-dub** [comment]
> Love this! I'd like to see an implementation that utilizes the native session API instead of saving the routes to file, which is how harpoon has been doing things, and it proved to be slow at scale.

After reading about sessions I think something like the following would be just as efficient at it is dead simple.

We have a `.snipe` file which we could check for in the current directory or recursively upwards. It holds contents like this:

```
lua/snipe/menu.lua:0:3
README.md:3:10
```

Just have to figure out how to save cursor position and output it like this.

**Comment by radgeRayden** [comment]
On a related note, I think it would be useful to allow this for session-specific pins as well, ie. without saving. Currently I implemented this by overriding the `get_sorted_buffer_list` to return only entries that match a list of pinned buffers. This would have been easier if I could register a filter for this sorted list, which could be useful to implement other functionality that I didn't think of.

**Comment by CTXO** [comment]
I don't think it would be redundant. On the projects that I work, for example, I have a set of files that I occasionally visit (e.g. settings file). It would help a lot if I could manually assign a persistent tag that I can easily remember for those files (like "s" for the settings file). And I don't think the "persist tags" feature can accomplish that. Maybe your idea of a different menu would make more sense for this.
